org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xerces/xmlmodel/XMLModelHandler.java

```diff
@@ -47,11 +53,28 @@
  */
 public class XMLModelHandler implements XMLComponent, XMLDocumentFilter {
 
+	private static final Logger LOGGER = Logger.getLogger(XMLModelHandler.class.getName());
+
 	private static final String VALIDATION = Constants.SAX_FEATURE_PREFIX + Constants.VALIDATION_FEATURE;
 
 	private static final String PARSER_SETTINGS = Constants.XERCES_FEATURE_PREFIX + Constants.PARSER_SETTINGS;
 
+	private static final List<XMLModelValidatorFactory> MODEL_VALIDATOR_FACTORIES;
+
+	static {
+		MODEL_VALIDATOR_FACTORIES = new ArrayList<>();
+		Iterator<XMLModelValidatorFactory> factories = ServiceLoader.load(XMLModelValidatorFactory.class).iterator();

```

Comment:
In some places, the different validators are loaded dynamically, but in others, like `LSPErrorReporterForXML`, they are hard coded. I think we should eventually get a clean separation between the different validators. This would let us do things like add a new validator that uses the lemminx grammar cache using the extension point mechanism, or compile a version of LemMinX with only specific validators in order to reduce the size.

However, I think maybe it makes more sense to do this in a future PR?

Reply:
> However, I think maybe it makes more sense to do this in a future PR?

I agree it is not perfect but XSD and DTD validator does that too. Please do that in an another PR.